### DISCUSSION TITLE
Portability of `ADIOS2` output files between different platforms.

### DISCUSSION THREAD

**Discussion by francesco-ballarin** [ornladios/ADIOS2#3990] [Q&A] [answered]
Dear `ADIOS2` community,
I am a beginner in `ADIOS2`, and with this question I would like to assess whether I can expect to be able to read on (say) a `arm64` platform output files generated by `ADIOS2` on a `amd64` one.

Further details to my query are:
* I searched for keywords like "portable" and "portability" in this repository and on the website, but could only find those words associated with the code itself, rather than output files. The [one instance](https://github.com/ornladios/ADIOS2/issues/2620) where output files were clearly mentioned was a negative one.
* my output format is either `BP4` or `BP5`. C++ types I am willing to save are mainly strings, integers and floats.
* I do not call `ADIOS2` myself, but I do so through a library that I use. If the way they call `ADIOS2` is relevant in answering the question, and you are willing to skim through a couple of files to see how they use `ADIOS2`, these are the relevant files: [file 1](https://github.com/FEniCS/dolfinx/blob/main/cpp/dolfinx/io/ADIOS2Writers.h), [file 2](https://github.com/FEniCS/dolfinx/blob/main/cpp/dolfinx/io/ADIOS2Writers.cpp) and [file 3](https://github.com/jorgensd/adios4dolfinx/blob/main/src/adios4dolfinx/checkpointing.py)

Thanks :)

**Comment by williamfgc** [comment, marked as answer]
BP files are binary and use standard C++ types, so in principle it's portable. As you saw some gotchas can happen with types that can change their underlying alignment. I would say give it a try and report back if there are issues. It's a common use case to write/read the data on different architectures (e.g. Summit). Hope it helps.

**Reply by francesco-ballarin** [reply]
Thanks @williamfgc. I'll give it a try then, and open an issue in case something goes wrong.
